av1/av1codecconfigurationrecord.go

```diff
@@ -0,0 +1,99 @@
+package av1
+
+import (
+	"io"
+
+	"github.com/Eyevinn/mp4ff/bits"
+)
+
+// DecConfRec - AV1CodecConfigurationRecord
+// Specified in https://github.com/AOMediaCodec/av1-isobmff/releases/tag/v1.2.0
+type DecConfRec struct {
+	Marker                           byte
+	Version                          byte
+	SeqProfile                       byte
+	SeqLevelIdx0                     byte
+	SeqTier0                         byte
+	HighBitdepth                     byte
+	TwelveBit                        byte
+	MonoChrome                       byte
+	ChromaSubsamplingX               byte
+	ChromaSubsamplingY               byte
+	ChromaSamplePosition             byte
+	Reserved                         byte
+	InitialPresentationDelayPresent  byte
+	InitialPresentationDelayMinusOne byte
+	ConfigOBUs                       []byte
+}
+
+// DecodeAVCDecConfRec - decode an AV1DecConfRec
+func DecodeAV1DecConfRec(data []byte) (DecConfRec, error) {
+	av1drc := DecConfRec{}
+
+	av1drc.Marker = data[0] >> 7
+	av1drc.Version = data[0] & 0x7F
+	av1drc.SeqProfile = data[1] >> 5
+	av1drc.SeqLevelIdx0 = data[1] & 0x1F
+	av1drc.SeqTier0 = data[2] >> 7
+	av1drc.HighBitdepth = (data[2] >> 6) & 0x01
+	av1drc.TwelveBit = (data[2] >> 5) & 0x01
+	av1drc.MonoChrome = (data[2] >> 4) & 0x01
+	av1drc.ChromaSubsamplingX = (data[2] >> 3) & 0x01
+	av1drc.ChromaSubsamplingY = (data[2] >> 2) & 0x01
+	av1drc.ChromaSamplePosition = data[2] & 0x03
+	av1drc.Reserved = data[3] >> 5
```

Comment:
I don't think we should save these under the name Reserved, because there may be another 4 reserved bits in the same byte.
I think we should rather report an error if these reserved bits are non-zero.